src/main/java/engine/movegen/PawnMovesGen.java

```diff
@@ -0,0 +1,92 @@
+package engine.movegen;
+
+import java.util.List;
+
+import engine.move.*;
+import engine.entities.Board;
+import engine.entities.Piece;
+
+public class PawnMovesGen extends MovesGenerator {
+    /**
+     * Creates list of possible moves a pawn can
+     * make which is later refined by PostMoveValidator.
+     * Checks if pawn can move one square, two squares,
+     * capture diagonally, promote, or en passant and adds
+     * them to the list of moves. Any moves which jump over
+     * a piece, are not on the board, or are occupied are
+     * removed from the list. King checks are NOT considered.
+     */
+    public List<Move> generate (Board board, Piece pawn) {
```

Comment:
Current `generate` method is too complex with too many branches. Consider adding helper functions to increase readability and to assist with future debugging.

Reply:
Please open an issue which says that this method should be refactored and better documented for sake of readability. We will not fix in this PR